File: sansa-query/sansa-query-spark/src/main/java/net/sansa_stack/query/spark/sparqlify/ResultSetSparkImpl.java
package net.sansa_stack.query.spark.sparqlify;

import org.apache.jena.sparql.core.Var;
import org.apache.jena.sparql.engine.binding.Binding;
import org.apache.spark.api.java.JavaRDD;

import java.util.List;

public class ResultSetSparkImpl
    implements JavaResultSetSpark
{
    protected List<Var> resultVars;
    protected JavaRDD<Binding> rdd;

    public ResultSetSparkImpl(List<Var> resultVars, JavaRDD<Binding> rdd) {
        super();
        this.resultVars = resultVars;
        this.rdd = rdd;
    }

    @Override public List<Var> getResultVars() {
        return resultVars;
    }
    @Override public JavaRDD<Binding> getRdd() {
        return rdd;
    }
}

File: sansa-sabine/src/main/scala/net/sansa_stack/query/spark/api/impl/ResultSetSparkImpl.scala
package net.sansa_stack.query.spark.api.impl

import net.sansa_stack.query.spark.api.domain.ResultSetSpark
import org.apache.jena.sparql.core.Var
import org.apache.jena.sparql.engine.binding.Binding
import org.apache.spark.rdd.RDD

/**
 * @author Lorenz Buehmann
 */
class ResultSetSparkImpl(resultVars: Seq[Var],
                         bindings: RDD[Binding])
  extends ResultSetSpark {
  override def getResultVars: Seq[Var] = resultVars

  override def getBindings: RDD[Binding] = bindings
}
